### PR TITLE
152772/create contact controller

### PR DESCRIPTION
**Pull request by jonathanNieto**
# Related tasks
+ [Creación de controladores y vistas básicas para interacción con el usuario](http://manoderecha.net/md/index.php/task/152772)

# Problem description
+ We need to create a controller  for Contact domain as well as the code for saving the information in the DB.
+ We need to create the necessary views  to support this functionality.

# Solution description
+ Added Contact controller
+ Added Contact views

# Test changes
+ Updated tests for Contact
+ Added tests for ContactController

# Evidence
+ Index Contact view
image
+ Show Contact view
image
+ Create contact view
image

# Testing
+ [X] Ran automated tests
+ [ ] Ran codenarc

# Database changes
+ None

### CODE REVIEW

File: grails-app/controllers/mephistopheles/ContactController.groovy [hunk 1]
```diff
@@ -2,5 +2,80 @@ package mephistopheles
 
 class ContactController {
 
-    static  scaffold = true
+    //static  scaffold = true
```
Comment (ernesto-castelan) [inline grails-app/controllers/mephistopheles/ContactController.groovy]: No olvides borrar el código que ya no se usa.

File: grails-app/domain/mephistopheles/Contact.groovy [hunk 2]
```diff
@@ -7,20 +7,20 @@ class Contact {
...
         mobilePhone matches: "[0-9]+", size: 8..12
         homePhone matches: "[0-9]+", size: 8..12
-        relationShip size: 3..25
+        relationship size: 3..25
     }
 
     @Override
     String toString() {
-        return name + " " + fatherLastName + " " + relationShip
+        return "$name $fatherLastName $relationship"
```
Comment (ernesto-castelan) [inline grails-app/domain/mephistopheles/Contact.groovy:23]: Los GroovyString son geniales! :D

File: test/unit/mephistopheles/ContactControllerSpec.groovy [hunk 3]
```diff
@@ -0,0 +1,394 @@
...
+        params.motherLastName = "García"
+        params.homePhone = "58745210"
+        params.mobilePhone = "5511223344"
+        params.relationship =  "Padre"
+
+        when: "Calling the save action"
+        controller.save()
+
+        /*then: "Success message is displayed"
+        flash.message == 'contact.created.message'*/
```
Comment (ernesto-castelan) [inline test/unit/mephistopheles/ContactControllerSpec.groovy]: Aquí igual, quitemos el código que no se usa.

Comment (jonathanNieto) [reply test/unit/mephistopheles/ContactControllerSpec.groovy]: :worried:

File: grails-app/views/contact/_errors.gsp [hunk 4]
```diff
@@ -0,0 +1,10 @@
+<g:hasErrors>
+    <div class="container alert alert-danger" role="alert">
+        <h5 class="display5">Por favor verifica los siguientes errores:</h5>
```
Comment (leonlipe) [inline grails-app/views/contact/_errors.gsp:3]: Acá es conveniente usar el message bundle.

File: grails-app/views/contact/_form.gsp [hunk 5]
```diff
@@ -0,0 +1,29 @@
+    <div class="form-group row">
+        <label for="name" class="col-sm-2 col-form-label">Nombre:</label>
+        <g:textField class="form-control col-sm-9"  name="name" maxlength="30" required="" value="${contactInstance?.name}"/>
+    </div>
+
+    <div class="form-group row">
+        <label for="fatherLastName" class="col-sm-2 col-form-label">Apellido Paterno:</label>
```
Comment (leonlipe) [inline grails-app/views/contact/_form.gsp]: El mismo tema del message...

**Pull request merged**